### PR TITLE
removing error code drop down

### PR DESCRIPTION
**Pull request by infinitydeep**
removing drop down for error codes, this MR means it should be automatically filled where possible instead.

### CODE REVIEW

File: src/js/contact-page.js [hunk 1]
```diff
@@ -21,45 +20,29 @@
...
     const { account_id = '' } = Utils.getJWTCookie('id_token') || {};
     return account_id;
   }
 
-  function getErrorCodeValues(formData) {
-    const error = formData.get('errorcode');
-    const errorText = document.querySelector(`option[value=${error}`).text;
-    const [errorCode, errorTitle] = errorText.split(': ');
-    return { errorCode, errorTitle };
+  function getLinks() {
```
Comment (Olie440) [inline src/js/contact-page.js:30]: I would rename this to be something like `getErrorCode`

File: src/js/contact-page.js [hunk 2]
```diff
@@ -21,45 +20,29 @@
...
+    const selectedError = ContactUsLinks.find(x => x.errorCode == errorCode);
+    return selectedError || { errorCode: 'UNKNOWN', linkText: 'UNKNOWN' };
   }
 
   function sendSupportEmail(formData) {
     const account_id = getAccountId();
-    const { errorCode, errorTitle } = getErrorCodeValues(formData);
+    const { errorCode, linkText } = getLinks();
```
Comment (Olie440) [inline src/js/contact-page.js:37]: Maybe these could be renamed `code` and `description` you can find the code that creates the object here: https://github.com/NHSDigital/nhs-login-help-centre/blob/develop/lib/article.js#L45

File: src/js/contact-page.js [hunk 3]
```diff
@@ -21,45 +20,29 @@
...
-      const firstOption = document.querySelector('#errorcode option');
-      firstOption.value = errorCode;
-      firstOption.hidden = false;
-      firstOption.innerText = errorDesc
-        ? `${errorCode}: ${decodeURIComponent(errorDesc)}`
-        : errorCode;
-    }
-    document.querySelector('#errorcode').value = errorCode;
-  }
+  const errorCode = Utils.getParam('error') || 'Unknown';
```
Comment (Olie440) [inline src/js/contact-page.js:23]: I don't think we need this since we have an or on the find

**Review by Olie440 — CHANGES_REQUESTED** [review]

File: src/js/contact-page.js (hunk 3 above)
Comment (Olie440) [reply src/js/contact-page.js:23]: I think as well it could probs be moved into the function as well

File: src/js/contact-page.js (hunk 1 above)
Comment (infinitydeep) [reply src/js/contact-page.js]: done

File: src/js/contact-page.js (hunk 2 above)
Comment (infinitydeep) [reply src/js/contact-page.js]: done

File: src/js/contact-page.js (hunk 3 above)
Comment (infinitydeep) [reply src/js/contact-page.js]: done

File: src/js/contact-page.js [hunk 4]
```diff
@@ -22,44 +22,34 @@
...
     return account_id;
   }
 
-  function getErrorCodeValues(formData) {
-    const error = formData.get('errorcode');
-    const errorText = document.querySelector(`option[value=${error}`).text;
-    const [errorCode, errorTitle] = errorText.split(': ');
-    return { errorCode, errorTitle };
+  function getErrorCode() {
+    if (errorCodeRegex.test(errorCode)) {
```
Comment (Olie440) [inline src/js/contact-page.js:31]: This looks to be missing the description, I also think the logic is slightly off.
The logic should be: 
- test if its a valid error code
- if it isn't return `{ code: 'UNKNOWN', description: 'UNKNOWN' }`
- if it is search for it
- if theres a match return the match
- if theres not a match then return the code and description from the getParams (defaulting to unknown if missing)

**Review by Olie440 — CHANGES_REQUESTED** [review]

File: src/js/contact-page.js [hunk 5]
```diff
@@ -21,45 +21,40 @@
...
+    const errorCode = Utils.getParam('error');
+    const errorDesc = Utils.getParam('desc');
+    if (errorCodeRegex.test(errorCode)) {
+      const errorMatch = ContactUsLinks.find(x => x.code == errorCode);
+      if (errorMatch) {
+        return errorMatch;
+      }
+      if (!errorMatch) {
+        return (
+          { code: errorCode, description: errorDesc } || { code: 'UNKNOWN', description: 'UNKNOWN' }
```
Comment (Olie440) [inline src/js/contact-page.js:39]: This would never return the second one since the you are declairing the object here. 
Looking at how the code is set up you could do this:
```js
const errorDesc = Utils.getParam('desc') || 'UNKNOWN';
// ....
return  { code: errorCode, description: errorDesc };

```

File: src/js/contact-page.js [hunk 6]
```diff
@@ -21,45 +21,40 @@
...
+        );
+      }
+    } else {
```
Comment (Olie440) [inline src/js/contact-page.js:42]: Since this is an edge case it should be an early return like this:
```js
if (!errorCodeRegex.test(errorCode)) {
   return { code: 'UNKNOWN', description: 'UNKNOWN' };
}
```

File: src/js/contact-page.js [hunk 7]
```diff
@@ -21,45 +21,40 @@
...
-    const error = formData.get('errorcode');
-    const errorText = document.querySelector(`option[value=${error}`).text;
-    const [errorCode, errorTitle] = errorText.split(': ');
-    return { errorCode, errorTitle };
+  function getErrorCode() {
+    const errorCode = Utils.getParam('error');
+    const errorDesc = Utils.getParam('desc');
+    if (errorCodeRegex.test(errorCode)) {
+      const errorMatch = ContactUsLinks.find(x => x.code == errorCode);
+      if (errorMatch) {
```
Comment (Olie440) [inline src/js/contact-page.js:34]: This could be an or statment like before:
```js
return errorMatch || { code: errorCode, description: errorDesc };
```

**Review by Olie440 — CHANGES_REQUESTED** [review]

File: src/js/contact-page.js [hunk 8]
```diff
@@ -28,20 +28,17 @@
...
-          { code: errorCode, description: errorDesc } || { code: 'UNKNOWN', description: 'UNKNOWN' }
-        );
-      }
-    } else {
+    const errorDesc = Utils.getParam('desc') || 'UNKNOWN';
+    const errorMatch = ContactUsLinks.find(x => x.code == errorCode);
+    if (!errorCodeRegex.test(errorCode)) {
       return { code: 'UNKNOWN', description: 'UNKNOWN' };
     }
+    if (errorMatch) {
```
Comment (Olie440) [inline src/js/contact-page.js:36]: No need for these if you are using `||`. Think about it:
- if errorMatch is truthy it will go in here and then always return errorMatch since it is truthy
- if errorMatch is falsy it will go into the bottom one and return that object, meaning that the || is unneeded

**Review by Olie440 — APPROVED** [review]

**Review by Tomo-Dachi — APPROVED** [review]

**Comment by lps88** [comment]
approved by Alex Murphy (tomo-dachi)
